### PR TITLE
Allow to check out branches

### PR DESCRIPTION
**Pull request by mikegehard**
## Description
The git server currently lacks branch switching capabilities, limiting both LLMs and developers. This adds branch checkout so LLMs can help developers add new functionality in a new feature branch.

## Server Details

- Server: git
- Changes to: tools

## Motivation and Context
See https://github.com/modelcontextprotocol/servers/issues/147

## How Has This Been Tested?
Tested via the Claude macOS app.

## Breaking Changes
No

## Types of changes

- [ ] New MCP Server
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist

- [X] I have read the [MCP Protocol Documentation](https://modelcontextprotocol.io)
- [ ] My server follows MCP security best practices
- [ X] I have updated the server's README accordingly
- [ X] I have tested this with an LLM client
- [ X] My code follows the repository's style guidelines
- [ X] New and existing tests pass locally
- [ ] I have added appropriate error handling
- [ ] I have documented all environment variables and configuration options

### CODE REVIEW

File: src/git/tests/test_server.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    result = git_checkout(repo, "test-branch")
+
+    assert "Switched to branch 'test-branch'" in result
+    assert repo.active_branch.name == "test-branch"
+
+def test_git_checkout_nonexistent_branch(tmp_path: Path):
+    repo = git.Repo.init(tmp_path)
+
+    with pytest.raises(git.GitCommandError):
+        git_checkout(repo, "nonexistent-branch")
```
Comment (dsp-ant) [inline src/git/tests/test_server.py:24]: I am not sure we should add this test. It feels straight forward enough.

If we do add tests, we should use a proper fixture:
```python
@pytest.fixture
def test_repository():
    # Setup
    temp_path = Path("temp_test_repo")
    temp_path.mkdir(exist_ok=True)
    
    # Initialize your repository here
    # ...
    
    yield temp_path
    
    # Cleanup
    shutil.rmtree(temp_path)

def test_git_checkout_existing_branch(test_repository):
   ...
```

**Review by dsp-ant — CHANGES_REQUESTED** [review]

**Comment by dsp-ant** [comment]
Overall this looks good. There are some changes I'd love to see to the test file. We either use a fixture or delete the tests.

File: src/git/tests/test_server.py (hunk 1 above)
Comment (mikegehard) [reply src/git/tests/test_server.py]: @dsp-ant I think I've addressed your concern? Let me know what you think.

**Review by dsp-ant — APPROVED** [review]
This looks good. Thank you

**Pull request merged**